### PR TITLE
Tell usage of 'path' from 'filepath' and fix *path checking

### PR DESCRIPTION
**Pull request by wmark**

### CODE REVIEW

**Comment by wmark** [comment]
Almost all changes in *browse.go* beginning with line 277 are whitespace changes, or a changed order in sort/cookie checking.

File: middleware/browse/browse.go [hunk 1]
```diff
@@ -226,143 +229,150 @@ func directoryListing(files []os.FileInfo, r *http.Request, canGoUp bool, root s
...
-			// if there's no sorting values in the cookies, default to "name" and "asc"
-			if sortErr != nil {
-				listing.Sort = "name"
-			} else { // if we have values in the cookies, use them
-				listing.Sort = sortCookie.Value
-			}
-		} else { // save the query value of 'sort' and 'order' as cookies
-			http.SetCookie(w, &http.Cookie{Name: "sort", Value: listing.Sort, Path: "/"})
-			http.SetCookie(w, &http.Cookie{Name: "order", Value: listing.Order, Path: "/"})
+	// Get the query vales and store them in the Listing struct
```
Comment (weingart) [inline middleware/browse/browse.go]: s/vales/values/

**Comment by weingart** [comment]
I like the path cleanup.  Why the conversion to a pointer to browse.Config?

File: middleware/browse/browse.go (hunk 1 above)
Comment (wmark) [inline middleware/browse/browse.go]: That shows as mine, but is really due to a whitespace change.

**Comment by wmark** [comment]
```go vet``` or ```go lint``` complained about copying this.

I've left it that way because I have a fork with locks in Config and expect *browse* to use such as well.

**Comment by wmark** [comment]
Sorry for the lack of one-branch one-feature discipline: Since *cla-assistant* is preventing me from pushing directly it's omnibus branches.

File: middleware/browse/browse.go (hunk 1 above)
Comment (mholt) [inline middleware/browse/browse.go]: Can you fix it anyway while we're at it? :smile:

**Comment by mholt** [comment]
Really? go vet and golint weren't showing any errors for me... was it complaining about a copied mutex or something?

**Comment by wmark** [comment]
Now that I see it — it's likely been the mutex. Keep it as pointer anyway?

**Comment by mholt** [comment]
I didn't know there was a mutex in template. How come I don't see any such warnings?

**Comment by wmark** [comment]
It's because there in a forked version. ^^;

**Comment by mholt** [comment]
But the browse.Config type exists in the master branch too...

**Comment by wmark** [comment]
Addressed.

File: middleware/browse/browse.go [hunk 2]
```diff
@@ -226,143 +229,150 @@ func directoryListing(files []os.FileInfo, r *http.Request, canGoUp bool, root s
...
-				} else { // if the 'limit' query is empty, or has the wrong value, list everything
-					marsh, err = json.Marshal(listing.Items)
-				}
-				if err != nil {
-					return http.StatusInternalServerError, err
-				}
-			} else { // there's no 'limit' query, list them all
+	listing.applySort()
+
+	var buf bytes.Buffer
```
Comment (weingart) [inline middleware/browse/browse.go]: I'd pull this out into its own function and make the if part look more like the else part with the template.

**Comment by weingart** [comment]
lgtm to me otherwise.  Definitely much cleaner than before.  :)

**Pull request merged**